stream_buffer.c

```diff
@@ -340,27 +339,19 @@ static void prvInitialiseNewStreamBuffer( StreamBuffer_t * const pxStreamBuffer,
             } /*lint !e529 xSize is referenced is configASSERT() is defined. */
         #endif /* configASSERT_DEFINED */
 
-        if( ( pucStreamBufferStorageArea != NULL ) && ( pxStaticStreamBuffer != NULL ) )

```

Comment:
Can you clarify why you want to remove the null pointer checks here?

Reply:
Check is redundant since there is an assertion for this at line 304 and 305.

Reply:
configASSERT is not always defined for production builds.

Reply:
Agree. Adding the NULL check here makes sense and it confirms to the API documentation as well. However I feel both assert and null check is redundant and adds to difference in API behavior  between production and non-production builds.

Reply:
Added `configASSERT` back to keep the current API behavior same.